src/math_ops.jl

```diff
@@ -263,17 +263,18 @@ julia> trace(dev(A))
 0.0
 ```
 """
-@generated function dev{dim}(S::SecondOrderTensor{dim})
-    Tt = get_base(S)
-    idx(i,j) = compute_index(Tt, i, j)
-    f = (i,j) -> i == j ? :((get_data(S)[$(idx(i,j))] - tr/3)) :
-                           :(get_data(S)[$(idx(i,j))])
-    exp = tensor_create(Tt, f)
-    return quote
-        $(Expr(:meta, :inline))
-        tr = trace(S)
-        $Tt($exp)
-    end
+@inline function dev(S::SecondOrderTensor)
+    Tt = get_base(typeof(S))
+    tr = trace(S) / 3
+    Tt(
+        @inline function(i, j)
```

Comment:
```julia
julia> Tensor{2, 3}((i, j)-> i == j ? 1 : 1.0)
ERROR: MethodError: Cannot `convert` an object of type Tuple{Int64,Float64,Float64,Float64,Int64,Float64,Float64,Float64,Int64} to an object of type Tensors.Tensor{2,3,T<:Real,M}
This may have arisen from a call to the constructor Tensors.Tensor{2,3,T<:Real,M}(...),
since type constructors fall back to convert methods.
 in Tensors.Tensor{2,3,T<:Real,M}(::Tuple{Int64,Float64,Float64,Float64,Int64,Float64,Float64,Float64,Int64}) at ./sysimg.jl:53
 in macro expansion at /home/fredrik/Dropbox/Programming/Tensors.jl/src/constructors.jl:15 [inlined]
 in Tensors.Tensor{2,3,T<:Real,M}(::##1#2) at /home/fredrik/Dropbox/Programming/Tensors.jl/src/constructors.jl:5
```